gcloud-java-dns/src/main/java/com/google/cloud/dns/testing/LocalDnsHelper.java

```diff
@@ -273,13 +281,18 @@ private String toJson(String message) throws IOException {
   private class RequestHandler implements HttpHandler {
 
     private Response pickHandler(HttpExchange exchange, CallRegex regex) {
-      URI relative = BASE_CONTEXT.relativize(exchange.getRequestURI());
+      URI relative = null;
+      try {
+        relative = BASE_CONTEXT.relativize(new URI(exchange.getRequestURI().getRawPath()));
```

Comment:
Oh, I wonder if that is correct... considering the example [here](https://cloud.google.com/storage/docs/json_api/v1/how-tos/batch).